src/harness/testcases/WINNF_FT_S_FAD_testcase.py

```diff
@@ -0,0 +1,182 @@
+          open(os.path.join('testcases', 'testdata', 'esc_sensor_record_0.json')))
+        self._sas_admin.InjectEscSensorDataRecord({'record': esc_sensor_record})
+        # STEP 3
+        self._sas_admin.TriggerFullActivityDump()
+        # STEP 4
+        response = self._sas.GetFullActivityDump()
+        # STEP 5
+        self.assertContainsRequiredFields("FullActivityDump.schema.json", response)
+        generation_date_time = datetime.strptime(response['generationDateTime'],
+                                          '%Y-%m-%dT%H:%M:%SZ')
+        # STEP 6 AND CHECK
+        if generation_date_time >= (datetime.utcnow() - timedelta(minutes=1)):
+            for activity_dump_file in response['files']:
+                # Verify the record type
+                # Verify the response files with ActivityDumpFile.schema.json Object schema
+                self.assertContainsRequiredFields("ActivityDumpFile.schema.json", activity_dump_file)
+                # Get json data from url
+                data = self._sas.DownloadFile(activity_dump_file['url'])
+                if activity_dump_file['recordType'] == 'cbsd':
+                    self.assertEqual(2, len(data['recordData']))
+                    # Verify that cbsds and grants in the full dump matches a cbsd or grant created acitivities
+                    for record in data['recordData']:
+                        # Verify the response files with CbsdData.schema.json Object schema
+                        self.assertContainsRequiredFields("CbsdData.schema.json", record)
+                        self.assertEqual(1, len(record['grants']))
+                        if record['registration']['fccId'] == device_a['fccId']:
+                            self.assertCbsdRecordEqual(record['registration'], device_a)
+                            self.assertGrantRecord(record['grants'][0], grant_a, grant_response[0])                   
+                        else:
+                            self.assertEqual(record['registration']['fccId'], device_c['fccId'])
+                            self.assertCbsdRecordEqual(record['registration'], device_c)
+                            self.assertGrantRecord(record['grants'][0], grant_c, grant_response[1])
+                if activity_dump_file['recordType'] == 'esc_sensor':
+                    self.assertEqual(1, len(data['recordData']))
+                    # Verify the response file of Esc Sensor
+                    self.assertContainsRequiredFields("EscSensorRecord.schema.json", data['recordData'][0])
+                    self.assertDictEqual(esc_sensor_record, data['recordData'][0])
+                if activity_dump_file['recordType'] == 'zone':
+                    self.assertEqual(1, len(data['recordData']))
+                    # Verify the response file of PPA
+                    self.assertContainsRequiredFields("zoneData.schema.json", data['recordData'][0])
+                    self.assertEqual(ppa_record['id'], data['recordData'][0]['id'])
```

Comment:
Can you use assertDictEqual() to make this code more compact?

Reply:
yes you are right, thank you, it is done.